contract/models/contract_line.py

```diff
@@ -41,6 +41,9 @@ class ContractLine(models.Model):
     last_date_invoiced = fields.Date(
         string='Last Date Invoiced', readonly=True, copy=False
     )
+    next_period_end_date = fields.Date(
+        string='Next Period End Date', copy=False
+    )
```

Comment:
I'd suggest adding a `next_period_start_date` too (computed = last_date_invoiced + 1). That will simplify the code a bit. And you can display to the user instead of `last_date_invoiced`, it will improve the UX.